course download button desktop

gumaerc opened the pull request:
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/929

#### What's this PR do?
This PR does the following:

 - Adds a "download course" button to the `/resources` page on every course with the URL being the base URL plus the course short id as the ZIP filename
 - Turns the resource lists into collapsible lists
 - Makes sure that the Lecture Videos are always on the top and expanded by default
 - Make sure other resource lists are collapsed by default

#### How should this be manually tested?
This PR is best tested by running `ocw-studio` locally and mocking out the publishing environment, so you can generate the ZIP files and the live course sites the same way they would be in RC and production:

 - Clone `ocw-studio` on the `cg/offline-build-short-id-zip-filename` branch
 - Make sure you've taken care of the following prerequisites:
   - Minio support configured
   - Concourse support configured
   - Custom Github organization configured
   - Some test courses in your database using the `ocw-course` starter, I'm using 9-40-introduction-to-neural-computation-spring-2018 in my example below but you should test any courses you want that have resources
   - In addition to configuring your `.env` for the above, also set:
```
RESOURCE_BASE_URL_DRAFT=https://draft.ocw.mit.edu
RESOURCE_BASE_URL_LIVE=https://live.ocw.mit.edu
STATIC_API_BASE_URL=https://live.ocw.mit.edu
OCW_HUGO_PROJECTS_BRANCH=main
OCW_HUGO_THEMES_BRANCH=cg/course-download
```
 - Make a temporary edit to `websites/views.py` to limit the sites seen by `mass-build-sites`:
```
        ...
        sites = Website.objects.filter(name="9-40-introduction-to-neural-computation-spring-2018")
        serializer = WebsiteMassBuildSerializer(instance=sites, many=True)
        return Response({"sites": serializer.data})
```
 - Bring up `ocw-studio` with `docker-compose up`
 - Run the following management commands one by one:
   - `./manage.py upsert_theme_assets_pipeline`
   - `./manage.py backpopulate_pipelines --filter 9-40-introduction-to-neural-computation-spring-2018`
   - `./manage.py upsert_mass_build_pipeline --offline --starter ocw-course-v2 --projects-branch main --themes-branch cg/course-download`
 - Go into the Concourse UI and unpause and run the following pipelines:
   - `ocw-theme-assets/main` - This is labeled as main but is overridden by our setting of `OCW_HUGO_THEMES_BRANCH`. Trigger a build and wait for it to finish.
   - In the `mass-build-sites` folder find the `live` pipeline matching the parameters we specified above when pushing the pipeline up. Trigger a build and wait for it to finish; it should only build the example course we specified above in our `views.py` override.
   - In the `live` folder, find the pipeline for `9-40-introduction-to-neural-computation-spring-2018` and trigger a build
 - Once the builds are complete, visit http://localhost:8045/courses/9-40-introduction-to-neural-computation-spring-2018/resources/ and verify the following:
   - The "Download course" button gives you a ZIP file from the offline build of the course
   - The other resource links should work, although the raw resources will be loaded from `live.ocw.mit.edu` because of our override above
   - The Lecture Videos section is on top and expanded on page load, with the other sections collapsed
   - The sections all expand and collapse as expected


#### Any background context you want to provide?
Note that the "Download all" buttons in the design on the collapsed sections are on hold until we introduce functionality that will generate these archives. This PR also only includes the desktop styles as the mobile design was not ready at the time of writing this PR.

#### Screenshots (if appropriate)
![chrome_2022-10-26_17-31-39](https://user-images.githubusercontent.com/12089658/198142378-1f0d729d-5919-4c8b-bd90-fbc1995d5ba9.png)
![chrome_2022-10-26_17-32-54](https://user-images.githubusercontent.com/12089658/198142571-c894c336-909b-4925-8f65-6acaef4d4ede.png)

ChristopherChudzicki reviewed (changes requested):
I ran through this with `wgs-s10-special-topics-in-women-gender-studies-seminar-latina-womens-voices-spring-2010` today and it worked well. Going to try one other course tomorrow morning that has some videos in it.

**UI Issue:** The expand/collapse chevrons appear swapped versus what I would expect and versus the designs (https://projects.invisionapp.com/share/PA131XTPEK76#/screens/470225774_Page_-_Download_Options_-_V1-1) So should update that.

gumaerc commented:
@ChristopherChudzicki Thanks for the review. I flipped the chevrons and changed the emphasis tags like you noted. I also pushed up another change that alters the filtering logic for the "expanded by default" video section. Previously it was looking only for "Lecture Videos" but I changed it to match anything with "Video" in the title, as I found a course during testing that just calls the section "Videos".

ChristopherChudzicki reviewed (approved):
@gumaerc This looks great.

As mentioned yesterday, I ran through the whole process with `wgs.s10-2010` and everything worked well.

**BUT:** I tried with `0.40-spring-2018` and keep getting this error:

Error: Error building site: failed to render pages: render of "home" failed: "/tmp/build/d1fb8792/ocw-hugo-themes/course-v2/layouts/home.html:18:32": execute of template failed: template: home.html:18:32: executing "subheader" at <partial "course_banner.html" .>: error calling partial: "/tmp/build/d1fb8792/ocw-hugo-themes/course-v2/layouts/partials/course_banner.html:6:137": execute of template failed: template: partials/course_banner.html:6:137: executing "partials/course_banner.html" at <delimit $courseData.level ", ">: error calling delimit: can't iterate over <nil>

(Running just `yarn start:course` in ocw-hugo-themes using that course + course-v2/config-offline.yaml works fine, although I don't get the static assets.)

I am going to approve, but can you double check that course?

gumaerc commented:
@ChristopherChudzicki A common issue here you might be experiencing is that the `release` branch of the course in question is not populated in your custom Github org. Can you check to make sure that it is? In the instructions I had you manually trigger the build pipeline but I probably should have had you do a publish  from the studio UI, which would trigger a push to the `release` branch on Github.

ChristopherChudzicki commented:
@gumaerc 👍 Worked when I did it through the UI
